docs/diagrams/CreateDoctorCommand-Logic.puml

```diff
@@ -0,0 +1,71 @@
+@startuml
+!include style.puml
+skinparam ArrowFontStyle plain
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":AddressBookParser" as AddressBookParser LOGIC_COLOR
+participant ":CreateDoctorCommandParser" as CreateDoctorCommandParser LOGIC_COLOR
+participant "d:CreateDoctorCommand" as CreateDoctorCommand LOGIC_COLOR
+participant "r:CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "m:Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("createDoctor XXX")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand("createDoctor XXX")
+activate AddressBookParser
+
+create CreateDoctorCommandParser
+AddressBookParser -> CreateDoctorCommandParser
+activate CreateDoctorCommandParser
+
+CreateDoctorCommandParser --> AddressBookParser
+deactivate CreateDoctorCommandParser
+
+AddressBookParser -> CreateDoctorCommandParser : parse("XXX")
+activate CreateDoctorCommandParser
+
+create CreateDoctorCommand
+CreateDoctorCommandParser -> CreateDoctorCommand
+activate CreateDoctorCommand
+
+CreateDoctorCommand --> CreateDoctorCommandParser :
+deactivate CreateDoctorCommand
+
+CreateDoctorCommandParser --> AddressBookParser : d
+deactivate CreateDoctorCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+CreateDoctorCommandParser -[hidden]-> AddressBookParser
+destroy CreateDoctorCommandParser
+
+AddressBookParser --> LogicManager : d
+deactivate AddressBookParser
+
+LogicManager -> CreateDoctorCommand : execute(Doctor)
```

Comment:
Here it should be `execute(m)` since we are passing in a model to the execute method instead of a Doctor object